Spring boot hw 2

ShemsheyMaxim opened the pull request.

ShemsheyMaxim committed:
added class Account,
added class Currency,
added javax dependency to pom.xml

ShemsheyMaxim committed:
added class Transaction,
added class TypeOperation

ShemsheyMaxim committed:
added interface AccountRepository,
added interface CurrencyRepository,
added interface AccountService and class AccountServiceImpl,
added interface CurrencyService and class CurrencyServiceImpl

ShemsheyMaxim committed:
added interface TransactionRepository,
added interface TypeOperationRepository,
added interface TransactionService and class TransactionServiceImpl,
added interface TypeOperationService and class TypeOperationServiceImpl

ShemsheyMaxim committed:
rewrote method injectData() in class InjectData for testing new implementations

KseniiaMakarova commented on src/main/java/com/springboot/onlinebanking/model/Currency.java:
```diff
@@ -0,0 +1,27 @@
+
+@Data
+@Entity
+@Table(name = "cyrrencies")
```
```suggestion
@Table(name = "currencies")
```

ShemsheyMaxim replied:
done

KseniiaMakarova commented on src/main/java/com/springboot/onlinebanking/model/Transaction.java:
```diff
@@ -0,0 +1,35 @@
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @OneToOne
```
why ManyToOne?

ShemsheyMaxim replied:
done

KseniiaMakarova commented on src/main/java/com/springboot/onlinebanking/model/Transaction.java:
```diff
@@ -0,0 +1,35 @@
+    @JoinColumn(name = "from_account", nullable = false)
+    private Account fromAccount;
+    @OneToOne
+    @JoinColumn(name = "to_account", nullable = false)
```
same

ShemsheyMaxim replied:
done

KseniiaMakarova commented on src/main/java/com/springboot/onlinebanking/service/AccountService.java:
```diff
@@ -0,0 +1,12 @@
+
+    List<Account> findAllUserAccountsByUserPhone(String phoneNumber);
+
+    Account findAccountByAccountNumber(String accountNumber);
```
```suggestion
    Account getByAccountNumber(String accountNumber);
```

ShemsheyMaxim replied:
done

KseniiaMakarova commented on src/main/java/com/springboot/onlinebanking/service/TransactionService.java:
```diff
@@ -0,0 +1,9 @@
+import java.util.List;
+
+public interface TransactionService {
+    List<Transaction> getAllByAccount(int page, int size, Account account);
```
let's also have a `save()` method

ShemsheyMaxim replied:
done

ShemsheyMaxim committed:
- renamed method from findAccountByAccountNumber() to getByAccountNumber in interfaces AccountRepository, AccountService and class AccountServiceImpl,
- renamed name table for Currency,
- changed relationship for field in Transaction class,
- added method save() for interface TransactionService and class TransactionServiceImpl
